src/Controller/BarcodeController.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use AppBundle\Entity\Delivery;
+use AppBundle\Entity\Task;
+use AppBundle\Service\TaskManager;
+use AppBundle\Utils\Barcode\BarcodeUtils;
+use Doctrine\Persistence\ManagerRegistry;
+use Picqer\Barcode\BarcodeGeneratorSVG;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Twig\Environment as TwigEnvironment;
+use libphonenumber\PhoneNumberFormat;
+use libphonenumber\PhoneNumberUtil;
+
+class BarcodeController extends AbstractController
+{
+    public function __construct(
+        private TwigEnvironment $twig,
+        private HttpClientInterface $browserlessClient,
+        private ManagerRegistry $doctrine,
+        private TaskManager $taskManager,
+        private BarcodeUtils $barcodeUtils,
+    )
+    { }
+
+
+
+    /**
+     * @Route("/api/barcode", name="barcode_api")
+     */
+    public function barcodeAction(
+        IriConverterInterface $iriConverter,
+        NormalizerInterface $normalizer,
+        Request $request
+    ): Response
+    {
+
+        if (!$request->get('code', null)) {
+           return $this->json(['error' => 'No code provided.'], 400);
+        }
+
+        $barcode = $this->barcodeUtils::parse($request->get('code'));
+
+        /** @var Task $task */
+        $task = $this->getDoctrine()
+            ->getRepository(Task::class)
+            ->findByBarcode($barcode->getRawBarcode());
+
+        if (is_null($task)) {
+            return $this->json(['error' => 'No data found.'], 404);
+        }
+
+        $client_action = null;
+        if ($task->isAssignedTo($this->getUser())) {
+            $client_action = 'ask_unassign';
+        } elseif ($task->isAssigned()) {
+            $client_action = 'ask_assign';
+        }
+
+        if (!$task->isAssignedTo($this->getUser())) {
+            if (is_null($task->getDelivery())) {
+                $task->assignTo($this->getUser());
+            } else {
+                $task->getDelivery()->assignTo($this->getUser());
+            }
+        }
+
+        $this->taskManager->scan($task);
```

Comment:
this will send the "TASK SCANNED" message through EDIFACT to the transporter? (see command handler)
in this case I am not 100% sure it is good to have such side effect here (because the endpoint is mostly about assign/unassign) so maybe move it to a different endpoint for separation of concern

tell me if i a misunderstood things